File: server/src/routes/artwork.ts
import { Router } from 'express';
import { upload } from '../middleware/upload';
import { uploadArtwork } from '../services/storageService';
import { prisma } from '../lib/prisma';
import { requireAuth } from '../middleware/requireAuth';

const router = Router();

router.post('/artworks', requireAuth, upload.single('image'), async (req, res) => {
  try {
    if (!req.file) {
      return res.status(400).json({ error: 'No image file provided'});
    }

    const imageUrl = await uploadArtwork(
      req.file.buffer,
      req.file.originalname,
      req.file.mimetype
    );
    
    const artwork = await prisma.artwork.create({
      data: {
        name: req.body.name,
        artist: req.body.artist,
        description: req.body.description,
        medium: req.body.medium,
        price: parseFloat(req.body.price),
        imageUrl,
        userId: (req as any).userId,
      },
    });

    res.status(201).json(artwork);
  } catch (error) {
    console.error(error);
    res.status(500).json({ error: 'Failed to create artwork'});
  }
});

export default router;

File: server/Models/Artwork.cs
namespace server.Models;

public class Artwork
{
	public int Id { get; set; }
	public string Title { get; set; } = string.Empty;
	public string Description { get; set; } = string.Empty;
	public decimal Price { get; set; } 
	public string ImageUrl { get; set; } = string.Empty;
	public int UserId { get; set; }
	public User? User { get; set; }
}